lib/bundler/cli/add.rb

```diff
@@ -2,23 +2,32 @@
 
 module Bundler
   class CLI::Add
-    def initialize(options, gem_name)
-      @gem_name = gem_name
+    def initialize(options, gems)
+      @gems = gems
       @options = options
       @options[:group] = @options[:group].split(",").map(&:strip) if !@options[:group].nil? && !@options[:group].empty?
     end
 
     def run
+      # raise error when no gems are specified
+      raise InvalidOption, "Please specify gems to add." if @gems.empty?
+
+      # show warning when options are used with mutiple gems
+      # for time being they will be applied to the all gems
+      Bundler.ui.warn "All the options will be applied to all gems`."
+
       version = @options[:version].nil? ? nil : @options[:version].split(",").map(&:strip)
 
       unless version.nil?
         version.each do |v|
           raise InvalidOption, "Invalid gem requirement pattern '#{v}'" unless Gem::Requirement::PATTERN =~ v.to_s
         end
       end
-      dependency = Bundler::Dependency.new(@gem_name, version, @options)
 
-      Injector.inject([dependency], :conservative_versioning => @options[:version].nil?) # Perform conservative versioning only when version is not specified
+      dependencies = @gems.each_with_object([]) {|g, d| d << Bundler::Dependency.new(g, version, @options) }
```

Comment:
ugh, this was added in Ruby 1.8.7
```ruby
dependencies = @gems.inject {|g, d| d << Bundler::Dependency.new(g, version, @options); d }
```
works according [to this](https://stackoverflow.com/questions/5481009/why-is-enumerableeach-with-object-deprecated) for 1.8.7

Though, I thought we had dropped support for old Rubies. 🤔

Reply:
I should have cross checked it 😓. My bad. 
Will fix it though.

Reply:
Won't be using a simple map be fine?
```ruby
dependencies = @gems.map {|g| Bundler::Dependency.new(g, version, @options) }
```

Reply:
that would work actually, yea!